pkgs/by-name/dp/dprint/plugins/default.nix

```diff
@@ -12,28 +16,26 @@ let
       license ? lib.licenses.mit,
       maintainers ? [ lib.maintainers.phanirithvij ],
     }:
-    fetchurl {
-      inherit
-        hash
-        url
-        pname
-        version
-        ;
-      name = "${pname}-${version}.wasm";
+    stdenv.mkDerivation (finalAttrs: {
```

Comment:
👍 for translating to mkDerivation, keeping the file extensions, and decoupling pname.

- It makes simple to add checkPhase when addressing issues like https://github.com/NixOS/nixpkgs/pull/370188
- Dprint plugins can provide json schema for configurations, using mkDerivation might include these files in future.

Reply:
> Is this required? As far as I know, the style of sha256 suffix only used for process plugins.

Yeah I just realised, since I am writing a `plugin.wasm` file, it can be used directly. no sha256 is required. so I will change it back to sri hash

Reply:
> sha256 suffix only used for process plugins

it did work for the main plugins (i.e. json, markdonw, typescript, toml, files did get properly formatted)

Reply:
It might be just ignored in WASM plugins.

```console
> dprint --version
dprint 0.49.1

> dprint check --plugins 'https://plugins.dprint.dev/json-0.20.0@malformed_hash_format'
https://plugins.dprint.dev/json-0.20.0@malformed_hash_format was specified as a plugin, but it doesn't look like one. Plugins must have a .wasm or .json extension.

> dprint check --plugins 'https://plugins.dprint.dev/json-0.20.0.wasm@malformed_hash_format'
```